File: R/Scripts/statistics/data distribution.R
data <- read.csv(file.path("Data","hw1_data.csv"))

percentiles <- quantile(data[["Temp"]], p= c(.05,.25,.5,.75,.95))

boxplot(data[["Temp"]])

breaks <- seq(from=min(data[["Temp"]]), to=max(data[["Temp"]]), length =5)
temp_freq <- cut(data[["Temp"]], breaks = breaks, right = TRUE, include.lowest = TRUE)
table(temp_freq) # frequency table

hist(data[["Temp"]], breaks = breaks)
         
hist(data[["Temp"]], freq = FALSE)
lines(density(data[["Temp"]]), lwd=3, col="green") # density plot

File: Python/statistics/exploratory_analysis/data distribution.py
import pandas as pd
import matplotlib.pyplot as plt

if __name__ == "__main__":
    fig_size = (12, 12)

    file = "../../data/statistics/hw1_data.csv"
    data = pd.read_csv(file)
    percentiles = data["Temp"].quantile([0.05, 0.25, 0.5, 0.75, 0.95])

    print(percentiles)

    # box plot
    ax = data["Temp"].plot.box(figsize=fig_size)
    plt.show()

    binned_temp = pd.cut(data["Temp"], 5)
    print(binned_temp.value_counts())  # frequency table

    # histogram
    ax = data["Temp"].plot.hist(figsize=fig_size)
    plt.show()

    # density plot
    ax = data["Temp"].plot.hist(density=True, figsize=fig_size)
    data["Temp"].plot.density(ax=ax)
    plt.show()
